### PR TITLE
[IMPROVEMENT] Checks for text before newlines on DVB subtitles

### PR DESCRIPTION
**Pull request by gray-v**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [X] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that another pull request for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [X] I have used CCExtractor just a couple of times.

---

Resolves the multiple lines issue of https://github.com/CCExtractor/ccextractor/issues/840, basically does the same thing that https://github.com/CCExtractor/ccextractor/pull/844 does but without raising potential issues with newlines as pointed out in the codein task.

Repeat of https://github.com/CCExtractor/ccextractor/pull/847 thank you github for automatically closing my PR for no reason

### CODE REVIEW

File: src/lib_ccx/ocr.c [hunk 1]
```diff
@@ -810,20 +810,39 @@ int compare_rect_by_ypos(const void*p1, const void *p2, void*arg)
...
+	while (*src != '\0')
+	{
+		//checks if a line has actual words in it before adding it
+		if (*src == '\n') {
+			line_scan = src - 1;
+			should_break = 1;
+			//continiously scans each line to find if there's a character in it. if so, don't break
+			while (*line_scan != '\0' && *line_scan != '\n') {
+				//mprint("%i %c\n", *line_scan, *line_scan);
+				if (*line_scan > 32 || line_scan < 127) { //ascii char codes
```
Comment (cfsmp3) [inline src/lib_ccx/ocr.c]: Why limit to ASCII? DVB is bitmap based - lots of non-ASCII characters in European subtitles.

File: src/lib_ccx/ocr.c [hunk 2]
```diff
@@ -810,20 +810,39 @@ int compare_rect_by_ypos(const void*p1, const void *p2, void*arg)
...
-	}
-	while(src != end + 1)
-	{
+	while (*src != '\0')
+	{
+		//checks if a line has actual words in it before adding it
+		if (*src == '\n') {
+			line_scan = src - 1;
+			should_break = 1;
+			//continiously scans each line to find if there's a character in it. if so, don't break
```
Comment (cfsmp3) [inline src/lib_ccx/ocr.c]: Typo

Comment (gray-v) [reply src/lib_ccx/ocr.c]: wait where's the typo i don't c it

**Comment by gray-v** [comment]
oh god nevermind i am way too sleepy to code this right now
will finish tomorrow

**Comment by cfsmp3** [comment]
Reopen after it's really well tested :-)

**Comment by gray-v** [comment]
I had a good sleep and tested independently with some test cases.

Original: "\nAAA\n\n\n   \n   \n   AA\n\n\n   A    A\n \n \n ."
Result: "\nAAA\n\n   AA\n   A    A\n ."

Original with the issue: 4 newlines on nodvbcolor, 3 without that option
Result after: 1 newline with, 2 without. Looks like nodvbcolor adds one with the </FONT> tags

**Pull request closed without merging**